jax/interpreters/pxla.py

```diff
@@ -1106,33 +1107,38 @@ def _pmap_sharding_spec(nrep, axis_size, npart, parts, sharded_aval, map_axis: O
   assert not ragged
   # get the sharding spec from inner sharded_jits as if we weren't in a pmap
   pspec = partitioned_sharding_spec(npart, parts, sharded_aval)
-  maybe_replicate = () if replication_factor == 1 else (Replicated(replication_factor),)
+  maybe_replicate = [] if replication_factor == 1 else [Replicated(replication_factor)]
   if map_axis is not None:
-    sharded_in_axis = sum(not isinstance(s, NoSharding) for s in pspec.sharding[:map_axis])
+    sharded_in_axis = sum(
+        not isinstance(s, NoSharding) for s in pspec.sharding[:map_axis])
+
     def shift_sharded_axis(a: MeshDimAssignment):
       if isinstance(a, ShardedAxis) and a.axis >= sharded_in_axis:
         return ShardedAxis(a.axis + 1)
       return a
     # replication_factor represents the product of inner pmaps, so it goes
     # after the outer pmapped axis at index 0
+    l = pspec.sharding
+    idx = map_axis
+    new_sharding = l[:idx] + [Unstacked(axis_size)] + l[idx:]
     return ShardingSpec(
-      sharding=tuple_insert(pspec.sharding, map_axis, Unstacked(axis_size)),
```

Comment:
tuple_insert does not work on lists, because it concanates a 1-element tuple:

def tuple_insert(t, idx, val):
  assert 0 <= idx <= len(t), (idx, len(t))
  return t[:idx] + (val,) + t[idx:] <<< This is invalid to concatenate a list and a tuple.